src/lib/swapper/swappers/ThorchainSwapper/utils/addAggregatorAndDestinationToMemo.test.ts

```diff
@@ -49,82 +37,83 @@ describe('addAggregatorAndDestinationToMemo', () => {
       finalAssetAddress: FINAL_ASSET_ADDRESS,
       minAmountOut,
       slippageBps,
-      finalAssetPrecision: 18,
-      chainNamespace: CHAIN_NAMESPACE.Evm,
+      longtailTokens: DEFAULT_THORCHAIN_ASSETS_LIST,
     })
 
     expect(modifiedMemo).toBe(
-      `=${MEMO_PART_DELIMITER}e${MEMO_PART_DELIMITER}${RECEIVE_ADDRESS}${MEMO_PART_DELIMITER}${expectedL1AmountOut}${MEMO_PART_DELIMITER}ss${MEMO_PART_DELIMITER}${affiliateBps}${MEMO_PART_DELIMITER}${AGGREGATOR_TWO_LAST_CHARS}${MEMO_PART_DELIMITER}${FINAL_ASSET_ADDRESS}${MEMO_PART_DELIMITER}206301621786412`,
+      `=${MEMO_PART_DELIMITER}e${MEMO_PART_DELIMITER}${RECEIVE_ADDRESS}${MEMO_PART_DELIMITER}${expectedL1AmountOut}${MEMO_PART_DELIMITER}ss${MEMO_PART_DELIMITER}${affiliateBps}${MEMO_PART_DELIMITER}${AGGREGATOR_TWO_LAST_CHARS}${MEMO_PART_DELIMITER}${SHORTENED_FINAL_ASSET_ADDRESS}${MEMO_PART_DELIMITER}941367142801`,
     )
   })
 
-  it('should throw if chainNamespace is UTXO and chainId is not BCH', () => {
+  it('should throw if chainId is BCH and bytes length is > 220', () => {
     const minAmountOut = '2083854765519275828179229'
-    const affiliateBps = '100'
-    const expectedL1AmountOut = '42' // we don't care about this for the purpose of tests
-    const quotedMemo = `=:ETH.ETH:${RECEIVE_ADDRESS}:${expectedL1AmountOut}:ss:${affiliateBps}:${AGGREGATOR_TWO_LAST_CHARS}:${FINAL_ASSET_ADDRESS}:10`
-
-    const slippageBps = 100 // 1%
+    const memoOver220Bytes = `=:ETH.ETH:${REALLY_BIG_ADDRESS}:${expectedL1AmountOut}:ss:${affiliateBps}`
 
     expect(() =>
       addAggregatorAndDestinationToMemo({
         sellChainId: dogeChainId,
-        quotedMemo,
+        quotedMemo: memoOver220Bytes,
         aggregator: AGGREGATOR_ADDRESS,
         finalAssetAddress: FINAL_ASSET_ADDRESS,
         minAmountOut,
         slippageBps,
-        finalAssetPrecision: 18,
-        chainNamespace: CHAIN_NAMESPACE.Utxo,
+        longtailTokens: DEFAULT_THORCHAIN_ASSETS_LIST,
       }),
     ).toThrow()
   })
 
-  it('should throw if chainId is BCH and bytes length is > 220', () => {
+  it('should throw if chainId is BTC and bytes length is > 80', () => {
     const minAmountOut = '2083854765519275828179229'
-    const affiliateBps = '100'
-    const expectedL1AmountOut = '42' // we don't care about this for the purpose of tests
-    const memoOver220Bytes = `=:ETH.ETH:${RECEIVE_ADDRESS}:${expectedL1AmountOut}:ss:${affiliateBps}:${AGGREGATOR_TWO_LAST_CHARS}:${FINAL_ASSET_ADDRESS}:${[
-      ...Array(220),
-    ].map((_, counter) => counter)}`
+    const memo = `=:ETH.ETH:${REALLY_BIG_ADDRESS}:${expectedL1AmountOut}:ss:${affiliateBps}`
+
+    expect(() =>
+      addAggregatorAndDestinationToMemo({
+        sellChainId: btcChainId,
+        quotedMemo: memo,
+        aggregator: AGGREGATOR_ADDRESS,
+        finalAssetAddress: FINAL_ASSET_ADDRESS,
+        minAmountOut,
+        slippageBps,
+        longtailTokens: DEFAULT_THORCHAIN_ASSETS_LIST,
+      }),
+    ).toThrow()
```

Comment:
Do we want to ensure throws throw with the expected error message? i.e not for a diff reason than memo size

Reply:
Actually it doesn't even go to the final memo size error as the exponent would require to be more than 2 numbers to reduce the length, added the exact error thrown, and I think due to to this assert, we can't check the latest assertion

Reply:
One last comment for paranoia: do we want to early assert/throw under this guy?
https://github.com/shapeshift/web/blob/5c6a1614077dc37c9ad5dcb30dc1bf49b5c7dd4f/src/lib/swapper/swappers/ThorchainSwapper/utils/addL1ToLongtailPartsToMemo.ts#L43

The current error we're testing against in tests is indicative of us probably wanting to check against max memo size earlier in the flow - i.e if the *original* memo is already over the limit, there's no point to continue

Reply:
Seems to me if the memo without aggregator and final asset amount out has already a length that's greater than the max memo size, wouldn't all further checks/stripping are effectively no-ops?

In other words, the intent/actual tests seems off:

- either we don't care about the pre-stripping length, and e.g stating that the memo is over 220 bytes wit `memoOver220Bytes` is useless (we're not really concerned about the *original* memo length being 220 here, since all we're testing is the memo length *after* adding shortened aggregator address and final amount out stripped)
- or we do, in this case, the error being thrown should reflect that

Reply:
Ohhhh got it now!

I still kept the assert I added so we throw earlier and avoid unneeded computations

And I added 2 more asserts:
- One super early when we receive the initial memo length, if it's over the limit we can already assert that we won't be able to validate it
- One right after adding the aggregator characters because we might encounter some cases where the aggregator makes it over the limit

And added a tests to test the aggregator case